x-pack/plugins/apm/server/routes/alerts/rule_types/anomaly/register_anomaly_rule_type.ts

```diff
@@ -144,12 +150,14 @@ export function registerAnomalyRuleType({
         }
 
         // start time must be at least 30, does like this to support rules created before this change where default was 15
-        const startTime = Math.min(
-          datemath.parse('now-30m')!.valueOf(),
+        const window =
+          datemath.parse('now-30m')!.valueOf() >
           datemath
-            .parse(`now-${ruleParams.windowSize}${ruleParams.windowUnit}`)
-            ?.valueOf() || 0
-        );
+            .parse(`now-${ruleParams.windowSize}${ruleParams.windowUnit}`)!
+            .valueOf()
+            ? '30m'
+            : `${ruleParams.windowSize}${ruleParams.windowUnit}`;
+        const { dateStart } = getTimeRange(window);
```

Comment:
nit
```suggestion
      // start time must be at least 30 min
      const dateStart = Math.max(
        getTimeRange("30m").dateStart,
        getTimeRange(`${ruleParams.windowSize}${ruleParams.windowUnit}`).dateStart
      );
```

Reply:
Yeah, sounds good. Didn't mean to block this.

Reply:
to you point @mikecote it should have been `Math.min` (and not `Math.max` as I wrote). That also means that the implementation is incorrect since it'll prefer the largest timestamp, which means we'll get a window of max 30min (we want the window to be **at least** 30 min).

I'll create a PR to fix this.